### PR TITLE
Default CLI mode to `-webSocket`

### PR DESCRIPTION
**Pull request by jglick**
https://github.com/jenkinsci/jenkins/pull/4369 introduced `-webSocket` transport, which was at the time considered experimental and so the default remained `-http`. Since then, as far as I know it has been stable (at least after #5455), so it seems appropriate to make it the default now rather than `-http` which relies on a hack (`FullDuplexHttpService`) to keep two HTTP connections open. Since both modes accept the same authentication option—namely, an API token—the change ought to be transparent for most users.

### Testing done

Ran with various transport options, or none, plus `-logger FINE` to confirm the mode being used.

### Proposed changelog entries

- The default connection mode for the Java CLI client is now `-webSocket`. You can specify `-http` to continue to use the former default (for example because you are running Jenkins in a servlet container other than the recommended built-in Jetty, or because you are running an unusual reverse proxy which does not support WebSocket). You can also continue to specify `-ssh` to use SSH transport (for example because you prefer to authenticate with a private key rather than an API token), or use a native SSH client.

### Proposed upgrade guidelines

If you use the Jenkins CLI but cannot make WebSocket connections to the Jenkins controller, you will now need to pass the `-http` or `-ssh` option if you were not already doing so.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7605"></a>

### CODE REVIEW

File: cli/src/main/java/hudson/cli/CLI.java
Comment (jglick) [inline cli/src/main/java/hudson/cli/CLI.java:127]: https://github.com/jenkinsci/jenkins/pull/4369#discussion_r362975529

If anything we could introduce something like

```bash
export JENKINS_CLI_OPTS="-http -s https://jenkins/ -auth admin:$(cat token)"
```

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by NotMyFault** [comment]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
